src/prefect/core/flow.py

```diff
@@ -160,6 +165,7 @@ def __init__(
         on_failure: Callable = None,
         validate: bool = None,
         result: Optional[Result] = None,
+        terminal_state_handler: Optional[Callable] = None,
```

Comment:
Might be worth including the signature of `[Flow, State, List[State]], Optional[State]` 🤷 since it's relatively simple